opwen_email_server/utils/email_parser.py

```diff
@@ -80,3 +83,28 @@ def _get_recipients(email: dict) -> Iterable[str]:
 def get_domains(email: dict) -> Iterable[str]:
     return frozenset(address.split('@')[-1]
                      for address in _get_recipients(email))
+
+
+def _get_as_base64(image_url):
+    res = requests.get(image_url)
+
+    if res.status_code != 200:
+        res.raise_for_status()
```

Comment:
If the image URL could not be fetched, this will raise an exception which will crash the queue worker that ingests the emails. The message will remain on the queue to be retried by the next worker which will crash again (and again until a maximum number of errors per email is hit). This is wasteful and will delay processing of the emails unnecessarily.

It would be better to just not modify the email in the case that there was an exception. You can handle this nicely by making this method return `Optional[str]` instead of `str` (and checking for the None return where the method is being called):

```py
if not res.ok:
    return None
```